plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/StaticSelector.java

```diff
@@ -137,22 +143,38 @@ private static void addNamedGroups(Pattern pattern, HashSet<String> variables)
         }
     }
 
-    private void addVariableValues(Pattern pattern, String candidate, Map<String, String> mapping)
+    /**
+     * @param optionalRegex The optional regex to match against the input.
+     * @param input The input to match against the regex.
+     * @param variables Variables to populate with the values from the regex.
+     * @return False iff the regex is present and the input does not match it, else true,
+     *         indicating matching should continue.
+     */
+    private boolean addVariablesForRegexIfMatching(Optional<Pattern> optionalRegex, String input, Map<String, String> variables)
     {
+        if (optionalRegex.isEmpty()) {
+            return true;
+        }
+        Pattern pattern = optionalRegex.get();
+        if (!pattern.matcher(input).matches()) {
```

Comment:
We can assign `pattern.matcher(input)` to a local variable and then use it in the loop below.

Reply:
Actually we can't; we need to reset the matcher on each iteration through the loop. I guess we could reuse this matcher on the _first_ iteration of the loop, and only reset the matcher on subsequent iterations, but that feels unnecessarily confusing for the benefit.

Reply:
Actually, upon further review, there's really no need for reseting the matcher here, you're right. I refactored this area a bit to remove some unnecessary logic and also remove the reliance on exception-handling. I also noticed that the variable logic wasn't tested at all in `TestStaticSelector`, so I updated that. Thanks for this callout!